### ISSUE TITLE
Save ProvisioningHierarchy in 1.5 creates some times wrong Xml-Element Order

### ISSUE THREAD

**Issue by czullu** [pnp/pnpframework#372]
the strange issue is, that saving the ProvisioningHierarchy with XMLOpenXMLTemplateProvider to a pnp-file creates sometimes the XML-Elements in wrong order which make the Template fail when load again in XmlPnPSchemaBaseSerializer.GetValidationResults

What did happen multiple time is that pnp:Templates is first followed by pnp:Preferences, pnp:Localization and the pnp:Teams

@PaoloPia i went back to commit e40396c4b3ef86d974ed39f0fd6426599a652cda - before schema changes. With that Version i can not reproduce the issue.

**Comment by jansenbe** [comment]
@czullu : I've seen that before and so far, we've not been able to reproduce. Our testing a while ago indicated that it only failed in .NET Core...are you using that?

**Comment by czullu** [comment]
Yes, It happens in .net core 3.1 for me and I was also not really able to track it down as it seems like 1 out of 10 having this issue.
I fixed it on my side by reorder the elements on import. Let me provide this and you can decide if that's good enough for now.

Holen Sie sich Outlook für Android<https://aka.ms/AAb9ysg>

________________________________

**Comment by andikrueger** [comment]
I'm facing this issue too. I see XML files with wrong sequences too>

```xml
  </pnp:Templates>
  <pnp:Preferences Generator="PnP.Framework, Version=1.6.0.0, Culture=neutral, PublicKeyToken=null" />
</pnp:Provisioning>
```

The xml above was created in a PowerShell 7.1 session using PnP.PowerShell to export a list as a template. The xml file is generated wrong in all my test runs.

Adding addtional data to the templates fails during schema validation with the following error: 
```powershell
Add-PnPDataRowsToSiteTemplate: 
Line |
  36 |      Add-PnPDataRowsToSiteTemplate -Path $tempFile -List 'PnPTestList' …
     |      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | The element 'Provisioning' in namespace 'http://schemas.dev.office.com/PnP/2021/03/ProvisioningSchema'
     | has invalid child element 'Preferences' in namespace
     | 'http://schemas.dev.office.com/PnP/2021/03/ProvisioningSchema'. List of possible elements expected:
     | 'Templates, Sequence, Teams, AzureActiveDirectory, Drive, ProvisioningWebhooks' in namespace
     | 'http://schemas.dev.office.com/PnP/2021/03/ProvisioningSchema'.

Add-PnPDataRowsToSiteTemplate:
Line |
  36 |      Add-PnPDataRowsToSiteTemplate -Path $tempFile -List 'PnPTestList' …
     |      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Invalid template file!

Write-Error: PnP.PowerShell.Commands.Model.PnPException
```

**Comment by andikrueger** [comment]
#450 is caused by this issue.
